src/Jitter_Optimize.cpp

```diff
@@ -1129,6 +1130,78 @@ bool CJitter::ConstantPropagation(StatementList& statements)
 	return changed;
 }
 
+bool CJitter::CopyPropagation2(StatementList& statements)
```

Comment:
Why do we need an extra pass for OP_SLL? Couldn't it be merged with the first one?

Reply:
`CopyPropagation2()` doesn't use an inner loop, it only checks for `OP_SLL` that's directly preceded by `OP_ADD`
if we used `CopyPropagation()` we could have a case where we find `OP_SLL` 4 instructions ahead of an `OP_ADD`, that would cause optimization to happen to only those 2 instructions, and `OP_SLL` wont apply to those inbetween.